packages/vscode-extension/lib/network/interceptors/PolyfillFetchInterceptor.ts

```diff
@@ -0,0 +1,433 @@
+  reject: (reason?: unknown) => void;
+}
+
+type DidCreateRequestFn = (requestId: number) => void;
+type DidReceiveNetworkResponseFn = (
+  requestId: number,
+  status: number,
+  headers: Record<string, string>,
+  url: string
+) => void;
+type DidReceiveNetworkIncrementalDataFn = (
+  requestId: number,
+  responseText: string,
+  progress: number,
+  total: number
+) => void;
+type DidReceiveNetworkDataFn = (requestId: number, response: BlobLikeResponse | string) => void;
+type DidCompleteNetworkResponseFn = (
+  requestId: number,
+  errorMessage: string,
+  didTimeOut: boolean
+) => void;
+
+interface PolyfillFetch {
+  // Internal properties
+  _nativeNetworkSubscriptions: Set<() => void>;
+  _nativeResponseType: NativeResponseType;
+  _nativeRequestHeaders: Record<string, string>;
+  _nativeResponseHeaders: Record<string, string>;
+  _nativeRequestTimeout: number;
+  _nativeResponse?: BlobLikeResponse | string;
+  _textEncoder: TextEncoder;
+  _requestId?: number;
+  _request: FetchRequest;
+  _response?: FetchResponse;
+  _streamController?: ReadableStreamDefaultController<Uint8Array>;
+  _stream?: ReadableStream<Uint8Array>;
+  _deferredPromise: DeferredPromise<Response>;
+  _responseStatus: number;
+  _responseUrl: string;
+  _abortFn?: () => void;
+}
+
+const LOADER_ID = "fetch-interceptor";
+const INITIATOR_TYPE = "script";
+const REQUEST_ID_PREFIX = "FETCH";
+
+/**
+ * Intercepts network requests made using the react-native-fetch-api polyfill package.
+ * https://github.com/react-native-community/fetch
+ *
+ * This interceptor hooks into the Fetch.js class internal lifecycle methods to capture
+ * network activity and report it via CDP-compliant messages (like XHRInterceptor). It handles three native
+ * response types:
+ * - "text": Streaming responses with incremental data callbacks
+ * - "blob": Single callback with Blob response (default)
+ * - "base64": Single callback with base64-encoded ArrayBuffer
+ *
+ * Should be used alongside XHRInterceptor to cover all network request types.
+ * Shares AsyncBoundedResponseBuffer with XHRInterceptor for response storage.
+ *
+ * @see polyfill_readme.md for detailed implementation notes
+ */
+class PolyfillFetchInterceptor {
+  private enabled: boolean = false;
+
+  private networkProxy?: NetworkProxy = undefined;
+  private responseBuffer?: AsyncBoundedResponseBuffer = undefined;
+
+  // timing
+  private startTime: number = 0;
+  private ttfbTime = 0;
+
+  private original__didCreateRequest: DidCreateRequestFn = () => {};
+  private original__didReceiveNetworkResponse: DidReceiveNetworkResponseFn = () => {};
+  private original__didReceiveNetworkIncrementalData: DidReceiveNetworkIncrementalDataFn = () => {};
+  private original__didReceiveNetworkData: DidReceiveNetworkDataFn = () => {};
+  private original__didCompleteNetworkResponse: DidCompleteNetworkResponseFn = () => {};
+  private original__abort: () => void = () => {};
+
+  private checkCompatibility() {
+    return !!Fetch;
```

Comment:
we should likely check if the internal methods from the prototype that we override exists here too